src/light_client.rs

```diff
@@ -387,4 +474,159 @@ mod tests {
+					)
+					.into(),
+					commitment: [
+						128, 34, 252, 194, 232, 229, 27, 124, 216, 33, 253, 23, 251, 126, 112, 244,
+						7, 231, 73, 242, 0, 20, 5, 116, 175, 104, 27, 50, 45, 111, 127, 123, 202,
+						255, 63, 192, 243, 236, 62, 75, 104, 86, 36, 198, 134, 27, 182, 224, 128,
+						34, 252, 194, 232, 229, 27, 124, 216, 33, 253, 23, 251, 126, 112, 244, 7,
+						231, 73, 242, 0, 20, 5, 116, 175, 104, 27, 50, 45, 111, 127, 123, 202, 255,
+						63, 192, 243, 236, 62, 75, 104, 86, 36, 198, 134, 27, 182, 224,
+					]
+					.to_vec(),
+				},
+				app_lookup: DataLookup {
+					size: 1,
+					index: vec![],
+				},
+			}),
+		};
+		let counter = Arc::new(Mutex::new(0u32));
+		let recv = Instant::now();
+		let kate_proof = [
+			Cell {
+				position: Position { row: 0, col: 2 },
+				content: [
+					183, 215, 10, 175, 218, 48, 236, 18, 30, 163, 215, 125, 205, 130, 176, 227,
+					133, 157, 194, 35, 153, 144, 141, 7, 208, 133, 170, 79, 27, 176, 202, 22, 111,
+					63, 107, 147, 93, 44, 82, 137, 78, 32, 161, 175, 214, 152, 125, 50, 247, 52,
+					138, 161, 52, 83, 193, 255, 17, 235, 98, 10, 88, 241, 25, 186, 3, 174, 139,
+					200, 128, 117, 255, 213, 200, 4, 46, 244, 219, 5, 131, 0,
+				],
+			},
+			Cell {
+				position: Position { row: 1, col: 1 },
+				content: [
+					172, 213, 85, 167, 89, 247, 11, 125, 149, 170, 217, 222, 86, 157, 11, 20, 154,
+					21, 173, 247, 193, 99, 189, 7, 225, 80, 156, 94, 83, 213, 217, 185, 113, 187,
+					112, 20, 170, 120, 50, 171, 52, 178, 209, 244, 158, 24, 129, 236, 83, 4, 110,
+					41, 9, 29, 26, 180, 156, 219, 69, 155, 148, 49, 78, 25, 165, 147, 150, 253,
+					251, 174, 49, 215, 191, 142, 169, 70, 17, 86, 218, 0,
+				],
+			},
+			Cell {
+				position: Position { row: 0, col: 3 },
+				content: [
+					132, 180, 92, 81, 128, 83, 245, 59, 206, 224, 200, 137, 236, 113, 109, 216,
+					161, 248, 236, 252, 252, 22, 140, 107, 203, 161, 33, 18, 100, 189, 157, 58, 7,
+					183, 146, 75, 57, 220, 84, 106, 203, 33, 142, 10, 130, 99, 90, 38, 85, 166,
+					211, 97, 111, 105, 21, 241, 123, 211, 193, 6, 254, 125, 169, 108, 252, 85, 49,
+					31, 54, 53, 79, 196, 5, 122, 206, 127, 226, 224, 70, 0,
+				],
+			},
+			Cell {
+				position: Position { row: 1, col: 3 },
+				content: [
+					132, 180, 92, 81, 128, 83, 245, 59, 206, 224, 200, 137, 236, 113, 109, 216,
+					161, 248, 236, 252, 252, 22, 140, 107, 203, 161, 33, 18, 100, 189, 157, 58, 7,
+					183, 146, 75, 57, 220, 84, 106, 203, 33, 142, 10, 130, 99, 90, 38, 85, 166,
+					211, 97, 111, 105, 21, 241, 123, 211, 193, 6, 254, 125, 169, 108, 252, 85, 49,
+					31, 54, 53, 79, 196, 5, 122, 206, 127, 226, 224, 70, 0,
+				],
+			},
+		]
+		.to_vec();
+		mock_client
+			.expect_fetch_cells_from_dht()
+			.returning(move |_, _| {
+				let fetched = cells_fetched.clone();
+				let unfetched = cells_unfetched.clone();
+				Box::pin(async move { Ok((fetched, unfetched)) })
+			});
+		if cfg.disable_rpc {
```

Comment:
Since we are creating default value for runtime configuration we should ensure that both paths are tested, or to explicitly test one path (e.g. by setting `cfg.disable_rpc` to `false` and leave only else branch of this condition.